src/config/user_profile.cpp

```diff
@@ -95,6 +122,13 @@ std::optional<std::chrono::seconds> UserProfile::get_nts_expiry() const {
 }
 
 void UserProfile::set_blinded_msgreqs(std::optional<bool> value) {
+    auto current_value = data["M"].exists()
+                               ? std::optional<bool>{static_cast<bool>(data["M"].integer_or(0))}
+                               : std::nullopt;
```

Comment:
The default value of an optional is nullopt, so you could perhaps make this slightly nicer as:
```suggestion
    std::optional<bool> current_value;
    if (data["M"].exists())
        current_value = static_cast<bool>(data["M"].integer_or(0));
```

Reply:
Updated